airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/CdcAcceptanceTests.java

```diff
@@ -251,15 +297,161 @@ public void testDeleteRecordCdcSync() throws Exception {
+
+    // We do not check that the source and the dest are in sync here because removing a stream doesn't
+    // delete its data in the destination
+    assertGlobalStateContainsStreams(connectionId, List.of(idAndNameStreamDescriptor));
+
+    LOGGER.info("Adding color palette stream back to configured catalog");
+    catalog = testHarness.discoverSourceSchema(sourceId);
+    LOGGER.info("Updated catalog: {}", catalog);
+    update = getUpdateInput(connectionRead, catalog, operationRead);
+    webBackendApi.webBackendUpdateConnection(update);
+
+    LOGGER.info("Waiting for sync job after update to start");
+    syncFromTheUpdate = waitUntilTheNextJobIsStarted(connectionId);
+    LOGGER.info("Checking that id_and_name table is unaffected by the partial reset");
+    assertDestinationMatches(ID_AND_NAME_TABLE, expectedIdAndNameRecords);
+    LOGGER.info("Checking that color_palette table was cleared in the destination due to the reset triggered by the update");
+    assertDestinationMatches(COLOR_PALETTE_TABLE, List.of());
+    LOGGER.info("Waiting for sync job after update to complete");
+    waitForSuccessfulJob(apiClient.getJobsApi(), syncFromTheUpdate);
+
+    // Verify that color palette table records exist in destination again after sync.
+    // If we see 0 records for this table in the destination, that means the CDC partial reset logic is
+    // not working properly, and it continued from the replication log cursor for this stream despite
+    // this stream's state being reset
+    assertDestinationMatches(COLOR_PALETTE_TABLE, expectedColorPaletteRecords);
+    assertGlobalStateContainsStreams(connectionId, List.of(idAndNameStreamDescriptor, colorPaletteStreamDescriptor));
+
+    // Verify that incremental still works properly after partial reset
+    LOGGER.info("Adding new records to tables");
+    final Instant beforeInsert = Instant.now();
+    source.query(ctx -> ctx.execute("INSERT INTO id_and_name(id, name) VALUES(6, 'geralt')"));
+    expectedIdAndNameRecords.add(new DestinationCdcRecordMatcher(
+        Jsons.jsonNode(ImmutableMap.builder().put(COLUMN_ID, 6).put(COLUMN_NAME, "geralt").build()),
+        beforeInsert,
+        Optional.empty()));
+
+    source.query(ctx -> ctx.execute("INSERT INTO color_palette(id, color) VALUES(4, 'yellow')"));
+    expectedColorPaletteRecords.add(new DestinationCdcRecordMatcher(
+        Jsons.jsonNode(ImmutableMap.builder().put(COLUMN_ID, 4).put(COLUMN_COLOR, "yellow").build()),
+        beforeInsert,
+        Optional.empty()));
+
+    LOGGER.info("Starting sync after insert");
+    final JobInfoRead connectionSyncRead2 = apiClient.getConnectionApi()
+        .syncConnection(new ConnectionIdRequestBody().connectionId(connectionId));
+    waitForSuccessfulJob(apiClient.getJobsApi(), connectionSyncRead2.getJob());
+
+    assertDestinationMatches(ID_AND_NAME_TABLE, expectedIdAndNameRecords);
+    assertDestinationMatches(COLOR_PALETTE_TABLE, expectedColorPaletteRecords);
```

Comment:
Is there a way we could assert only the records generated in the second sync?

Reply:
We could, but it feels slightly better to me to assert that the all of the records in the destination are as expected, rather than just the new ones, since this would fail in weird cases where maybe the earlier records were deleted for some reason, in which case we would not want to succeed.

I added the new records that are generated in the second sync to the `expectedXRecords` lists, and this assertion checks the `_ab_cdc_updated_at` column for those records to verify that that timestamp comes after the `beforeInsert` timestamp that was saved before the assertion, so we are still verifying that these new records came from this new sync specifically.